### DISCUSSION TITLE
Different media control results on different browsers

### DISCUSSION THREAD

**Discussion by lloydbayley** [mcguirepr89/BirdNET-Pi#777] [Q&A] [answered]
G'day!

I've noticed that there's some differences in media display between browsers. I hadn't thought of this before but after reading issue #759, I noticed a report in there and decided to try things.

On spectrogram recordings at the bottom, I get varying results.
I'm on a Mac, so:

Firefox yields

<img width="989" alt="image" src="https://user-images.githubusercontent.com/21309308/223422536-7452dab9-f890-426d-bd7e-e9cd4a0ca4b5.png">

Which I thought was normal and all works.

Google Chrome yields:

<img width="1009" alt="image" src="https://user-images.githubusercontent.com/21309308/223422738-415f3595-8544-42c8-9c0e-a652fe21c70d.png">

The volume and fullscreen icons seem to be either disabled or obfuscated by an overlay etc - no biggy.
Note that the bottom right three dots produces the following menu

<img width="1024" alt="image" src="https://user-images.githubusercontent.com/21309308/223423276-90f42de7-3a68-4525-99fd-4491000fdeea.png">

Which is interesting with new options!

However, Safari yields:

<img width="1038" alt="image" src="https://user-images.githubusercontent.com/21309308/223424310-4350d20f-7155-48df-bb41-dd58ae5398d8.png">

Nothing, basically. The play button still exists in the middle of the spectrogram, which is fine, however, no media controls at the bottom.

I was pondering the differences and just supposed that browsers do things differently, but the download and (perhaps less-so) playback speed controls might be a nice thing to have.

I'm a bit rusty on media webcontrols these days.
I'd be interested in your thoughts!

**Comment by ehpersonal38** [comment, marked as answer]
Hi Lloyd - thanks for looking into this. I always appreciate exploring the intricacies of different software functions. 

You're right that different browsers handle these things differently, and we developers don't really have much control over it unless we want to implement a custom video player plugin.

As for Safari, I think what's happening there is that the media controls aren't showing due to some code I implemented a few weeks ago that disables media controls if there are more than 100 detections displayed on the "Recordings" page. (d257b144a184f25e6346187e9983e76f4317c465) This is to reduce lag and overall make the "Recordings" page a lot more usable.

So if you want to see what Safari's media controls really look like I recommend taking a look at any of the detections on "Overview" or "Today's Detections" - those will definitely not be affected by my code which disables the controls.

Take care,
Ethan

**Reply by lloydbayley** [reply]
Agreed. It's not really a big deal. Was just more interested in the differences and if anyone had found anything different.
I absolutely agree that a special media module would be fraught with peril and keeping it up to date and inclined to problems.
Makes things interesting doesn't it?! :)
Thanks for confirming my thoughts.
